standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -656,10 +656,9 @@ public void createCatalog(Catalog cat) throws MetaException {
   }
 
   @Override
-  public void alterCatalog(String catName, Catalog cat)
-      throws MetaException, InvalidOperationException {
+  public void alterCatalog(String catName, Catalog cat) {
     if (!cat.getName().equals(catName)) {
-      throw new InvalidOperationException("You cannot change a catalog's name");
+      throw new HiveMetaRuntimeException("You cannot change a catalog's name: " + cat.getName() + " -> " + catName);
```

Comment:
Not a very enterprise-grade exception message. Should we re-word this to something like this? "Catalogs cannot be renamed"

Reply:
Should we distinguish between invalid operations/input arguments that are not permitted such as above vs runtime exceptions from backend DDL/DML operations?

Reply:
@nrg4878 `InvalidOperationException` is a Thrift exception, so I'm trying to get it out of this class.

I am open to also making this a POJO `IllegalArgumentException` as well, but this would apply to quite a few methods so it probably should be done in another PR.

Reply:
Thanks for the input on the error text. I've altered it a bit to make it more clear.